samples/crypto/test_sample/README.rst

```diff
@@ -0,0 +1,258 @@
+| HMAC               | HMAC SHA256 | `NIST - HMAC`_                                                             | `NIST - HMAC`_                                                             |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+|                    | HMAC SHA512 |                                                                            | `NIST - HMAC`_                                                             |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+| HKDF               | HKDF SHA256 | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+|                    | HKDF SHA512 |                                                                            | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+
+
+Requirements
+************
+
+* One of the following development boards:
+
+  * |nRF9160DK|
+  * |nRF5340DK|
+  * |nRF52840DK|
+  * |nRF52DK|
+  * |nRF51DK|
+
+* something else??
+
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/bluetooth/central_bas`
+
+.. include:: /includes/build_and_run.txt
+
+.. _crypto_test_testing:
+
+Testing
+=======
+
+1. Compile and program the application.
+#. Observe the result of the different test vectors in the log using an RTT viewer or a terminal emulator.
+   Make sure the test vector verdict is "passed"::
+
+      ***************************************
+      All Tests Done
+      X test vectors passed
+      0 test vectors failed
+      ***************************************
+#.
+
+Additional test cases and test vectors
+======================================
+
+Test cases and test vectors can be added to the test suite either by including additional source files or by extending the existing files.
+
+
+Test case
+---------
+
+A test case is a function designed to verify parts of the functionality of an nrf_crypto operation.
+Most nrf_crypto operations, like hash calculations and ECDH, have multiple test cases to be able to cover all features.
+A typical test case loops over all related test vectors, and logs the verdict of each.
```

Comment:
A typical test case is called by looping over the registered test vectors and calling the test case. The execution logs the verdic of each test vector

Reply:
done